OPHD/UI/NotificationArea.cpp

```diff
@@ -118,9 +118,23 @@ void NotificationArea::onMouseDown(EventHandler::MouseButton button, int x, int
 }
 
 
-void NotificationArea::onMouseMove(int x, int y, int dX, int dY)
+void NotificationArea::onMouseMove(int x, int y, int /*dX*/, int /*dY*/)
 {
+	if (!rect().contains({ x, y })) { return; }
 
+	size_t count = 0;
+	for (auto& rect : mNotificationRectList)
+	{
+		if (rect.contains({ x, y }))
+		{
+			mNotificationIndex = count;
+			return;
+		}
+
+		count++;
+	}

```

Comment:
Nothing wrong with this. Just wanted to mention it would be possible to use [`std::find_if`](https://en.cppreference.com/w/cpp/algorithm/find) and [`std::distance`](https://en.cppreference.com/w/cpp/iterator/distance) to compute this.

```cpp
	const auto contains = [point = Point<int>{x, y}](const Rectangle<int>& rect) { return rect.contains(point); };
	const auto iterator = std::find_if(mNotificationRectList.begin(), mNotificationRectList.end(), contains);
	if (iterator != mNotificationRectList.end()) {
		mNotificationIndex = std::distance(mNotificationRectList.begin(), iterator);
		// Rest of code for found case
	}
```

I think I prefer the readability of ranged-for loops over some STL algorithms. Mind you, with the Range changes coming in C++20, those STL algorithms should become must shorter and more readable, so I'm starting to think of how the STL can be used more. In particular, the Range changes should allow less repetition, as you can then pass collections instead of pairs of iterators, so no need for the `begin()` and `end()` pairs.